internal/app/donor_store.go

```diff
@@ -130,6 +135,13 @@ func (s *donorStore) Latest(ctx context.Context) (*page.Lpa, error) {
 }
 
 func (s *donorStore) Put(ctx context.Context, lpa *page.Lpa) error {
+	newHash, err := lpa.GenerateHash()
+	if newHash == lpa.Hash || err != nil {
+		return err
```

Comment:
For instances where `Lpa` has not changed should we not do anything rather than return an error? I can see you have a test for that but I'm not getting how it's passing if we return an error here for a matching hash?

Reply:
`err` will be `nil` if `GenerateHash` worked fine, this is just me combining two ifs, like:

```go
if err != nil {
  return err
}
if newHash == lpa.Hash {
  return nil // but also err==nil as the if above didn't return
}
```